bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/Clip2Bridge.java

```diff
@@ -0,0 +1,1049 @@
+     * the following events are also considered as fatal:
+     *
+     * <li>onClosed()</li>
+     * <li>onReset()</li>
+     */
+    private class EventStreamListenerAdapter extends BaseStreamListenerAdapter<Boolean> {
+        private final DataFrameCollector eventData = new DataFrameCollector();
+
+        @Override
+        public void onClosed(@Nullable Stream stream) {
+            handleHttp2Error(Http2Error.CLOSED);
+        }
+
+        @Override
+        public void onData(@Nullable Stream stream, @Nullable DataFrame frame, @Nullable Callback callback) {
+            Objects.requireNonNull(frame);
+            Objects.requireNonNull(callback);
+            synchronized (this) {
+                eventData.append(frame.getData());
+                BufferedReader reader = new BufferedReader(eventData.contentStreamReader());
+                @SuppressWarnings("null")
+                List<String> receivedLines = reader.lines().collect(Collectors.toList());
+
+                // a blank line marks the end of an SSE message
+                boolean endOfMessage = !receivedLines.isEmpty()
+                        && receivedLines.get(receivedLines.size() - 1).isBlank();
+
+                if (endOfMessage) {
+                    eventData.reset();
+                    // receipt of ANY message means the event stream is established
+                    if (!completable.isDone()) {
+                        completable.complete(Boolean.TRUE);
+                    }
+                    // append any 'data' field values to the event message
+                    StringBuilder eventContent = new StringBuilder();
+                    for (String receivedLine : receivedLines) {
+                        if (receivedLine.startsWith("data:")) {
+                            eventContent.append(receivedLine.substring(5).stripLeading());
+                        }
+                    }
+                    // note: StringBuilder.isEmpty() only available from Java 15+
+                    if (eventContent.length() > 0) {
+                        onEventData(eventContent.toString().trim());
+                    }
+                }
+            }
+            callback.succeeded();
+        }
+
+        @Override
+        public boolean onIdleTimeout(@Nullable Stream stream, @Nullable Throwable x) {
+            return false;
+        }
+
+        @Override
+        public void onReset(@Nullable Stream stream, @Nullable ResetFrame frame) {
+            handleHttp2Error(Http2Error.RESET);
+        }
+    }
+
+    /**
+     * Enum of potential fatal HTTP 2 session/stream errors.
+     */
+    private enum Http2Error {
+        CLOSED,
+        FAILURE,
+        TIMEOUT,
+        RESET,
+        IDLE,
+        GO_AWAY,
+        UNAUTHORIZED;
+    }
+
+    /**
+     * Private exception for handling HTTP 2 stream and session errors.
+     */
+    @SuppressWarnings("serial")
+    private static class Http2Exception extends ApiException {
```

Comment:
Since this file is rather large (> 1000 lines) consider extracting this class to another file.